tests/python/contrib/test_hexagon/pytest_util.py

```diff
@@ -109,25 +113,32 @@ def get_multitest_ids(
     ]
 
 
-def get_numpy_dtype_info(np_dtype_name: str) -> Union[np.finfo, np.iinfo]:
+def get_numpy_dtype_info(dtype: Union[str, np.dtype]) -> Union[np.finfo, np.iinfo]:
     """
     Return an appropriate 'np.iinfo' or 'np.finfo' object corresponding to
-    the specified dtype.
+    the specified Numpy dtype.
     """
-    np_dtype = np.dtype(np_dtype_name)
+
+    if type(dtype) == str:
+        np_dtype = np.dtype(np_dtype_name)
+    else:
+        assert isinstance(dtype, np.dtype)
```

Comment:
can you add an explainer when this assert fails:
```suggestion
        assert isinstance(dtype, np.dtype), f"dtype: want str or np.dtype, got {dtype!r}"
```